src/lang/sclang.js

```diff
@@ -310,11 +319,24 @@ export default class SCLang extends EventEmitter {
 
   makeStateWatcher(): SclangIO {
     let stateWatcher = new SclangIO(this);
-    for (let name of ['interpreterLoaded', 'error', 'stdout', 'state']) {
+    var alreadyReady = false; // have we switched to the ready state?
+    for (let name of ['error', 'stdout']) {
       stateWatcher.on(name, (...args) => {
         this.emit(name, ...args);
       });
     }
+    stateWatcher.on('state', newState => {
```

Comment:
I don't think that the stateWatcher event listener should be keeping track of changes. It is only supposed to re-emit the event. The consumer can keep track and ignore if it needs to.

The reason you had to put this in is because you added "CAPTURING" as a new state, but the intention of the states is only to track the major lifecycle of the `sclang` interpreter. 

As you have it now it would go back and forth from READY to CAPTURING.

I think you are assuming that there can only be one interpreter call active at a time. In fact, you can launch many of them and each one many do async things over in supercollider lang, and return results or errors at any point in the future.